eyeshade/lib/queries.js

```diff
@@ -65,3 +67,35 @@ function timeConstraintSettlements (options = {}) {
  order by paid ${order}
  limit $2;`
 }
+
+function referralGroups (options = {}) {
+  let { fields = [], active } = options
+  const fieldsMap = {
+    name: 'name',
+    amount: 'amount',
+    currency: 'currency',
+    codes: 'countries.codes AS codes'
+  }
+  fields = _.isString(fields) ? fields.split(',').map((str) => str.trim()) : fields
+  return `
+SELECT
+  id,
+  min_referral_time as "minReferralTime"
+  ${fields.reduce((memo, key) => {
+    const query = fieldsMap[key]
+    if (query) {
+      memo.push(',\n  ', query)
+    }
+    return memo
+  }, []).join('')}
```

Comment:
Would it be simpler to just request all the fields and return null if they don't exist than dynamically compute the query?

Reply:
i suppose there is no concern about the size of the response so yeah, i'd be fine with that